src/storage/src/table/state_table.rs

```diff
@@ -235,6 +274,221 @@ impl<S: StateStore> StateTableRowIter<S> {
+                        match row_op {
+                            RowOp::Insert(row) | RowOp::Update((_, row)) => {
+                                yield Cow::Borrowed(row);
+                            }
+                            _ => {}
+                        }
+                    }
+                }
+
+                (
+                    Some(Ok((cell_based_pk, cell_based_row))),
+                    Some(Ok((mem_table_pk, _mem_table_row_op))),
+                ) => {
+                    match cell_based_pk.cmp(mem_table_pk) {
+                        Ordering::Less => {
+                            // cell_based_table_item will be return
+                            let row: Row = cell_based_table_iter.next().await.unwrap()?.1;
+                            yield Cow::Owned(row);
+                        }
+                        Ordering::Equal => {
+                            // mem_table_item will be return, while both cell_based_streaming_iter
+                            // and mem_table_iter need to execute next()
+                            // once.
+                            let (mem_table_pk, row_op) = mem_table_iter.next().unwrap()?;
+                            if (start_key, end_key).contains(mem_table_pk) {
+                                match row_op {
+                                    RowOp::Insert(row) => yield Cow::Borrowed(row),
+                                    RowOp::Delete(_) => {}
+                                    RowOp::Update((old_row, new_row)) => {
+                                        debug_assert!(old_row == cell_based_row);
+                                        yield Cow::Borrowed(new_row);
+                                    }
+                                }
+                            }
+                            cell_based_table_iter.next().await.unwrap()?;
+                        }
+                        Ordering::Greater => {
+                            // mem_table_item will be return
+                            let (mem_table_pk, row_op) = mem_table_iter.next().unwrap()?;
+                            if (start_key, end_key).contains(mem_table_pk) {
+                                match row_op {
+                                    RowOp::Insert(row) => yield Cow::Borrowed(row),
+                                    RowOp::Delete(_) => {}
+                                    RowOp::Update(_) => unreachable!(),
+                                }
+                            }
+                        }
+                    }
+                }
+                (Some(_), Some(_)) => {
+                    // Throw the error.
+                    cell_based_table_iter.next().await.unwrap()?;
+                    mem_table_iter.next().unwrap()?;
+
+                    unreachable!()
+                }
+            }
+        }
+    }
+
+    #[try_stream(ok = Cow<'a, Row>, error = StorageError)]
+    async fn into_stream_with_prefix<'a>(
```

Comment:
Looks a little duplicated with `into_stream_with_bounds`. Is it possible to unify them into one? 🤔 (Prefix is also some kind of ranges, I think)

Reply:
Thanks for your comments and I have simplify the duplicated part.